rss.go

```diff
@@ -0,0 +1,153 @@
+package main
+
+import (
+	"fmt"
+	"log"
+	"regexp"
+	"strconv"
+	"time"
+
+	"github.com/mmcdole/gofeed"
+	ext "github.com/mmcdole/gofeed/extensions"
+)
+
+type Feed struct {
+		Username    string
+		DisplayName string
+		IconURL     string
+		Entries     []*FeedEntry
+}
+
+type FeedEntry struct {
+		ID          string
+		URL         string
+		Title       string
+		Year        string
+		Rating      int
+		WatchedDate time.Time
+		Rewatch     bool
+		Poster      string
+		Review      string
+}
+
+// Fetches a user's RSS feed, returning an array of 50 FeedEntrys with parsed values
+func GetFeed(username string) (Feed) {
+	fp := gofeed.NewParser()
+	feed, err := fp.ParseURL("https://letterboxd.com/" + username + "/rss/")
+
+	if err != nil {
+		log.Fatalf("failed to fetch feed: %v\n", err)
+	}
+
+	entries := []*FeedEntry{}
+	for i := 0; i < 50; i++ {
+		entries = append(entries, parseEntry(feed.Items[i]))
+	}
+
+	return Feed{
+		username,
+		feed.Items[0].Author.Name,
+		"iconURL",
+		entries,
+	}
+}
+
+func parseEntry(entry *gofeed.Item) (*FeedEntry) {
+	watchedDate := handleWatchedDate(entry.Extensions["letterboxd"]["watchedDate"])
+	poster, review := handleData(entry.Title, entry.Description, watchedDate)
+
+	return &FeedEntry{
+		entry.GUID,
+		entry.Link,
+		handleFilmTitle(entry.Extensions["letterboxd"]["filmTitle"]),
+		handleYear(entry.Extensions["letterboxd"]["filmYear"]),
+		handleRating(entry.Extensions["letterboxd"]["memberRating"]),
+		watchedDate,
+		handleRewatch(entry.Extensions["letterboxd"]["rewatch"]),
+		poster,
+		review,
+	}
```

Comment:
Since there's so many fields here, better to use their names and not just depend on the order. For example:
```go
return &FeedEntry{
	ID: entry.GUID,
	...
}
```